### DISCUSSION TITLE
pH and Chlorine sensor recommendations?

### DISCUSSION THREAD

**Discussion by MaxVonEvil** [tagyoureit/nodejs-poolController#843] [General]
Hi all, having wrestled my pump, valves, heating and temperature into submission by automation, I'm ready to take on the *chemistry*. First thing's first, I got to lay my hands on some sensors and get them plumbed into the discharge line. Looking for recommendations and experiences on the hardware 

Having looked around, it seems sensors come in all ranges, from pure chinesium alloy to astronomically expensive sensors. I guess what I'm looking for are middle-of-the-road sensors with a reasonable lifespan which are relatively easy to replace by way of the pipe adaptor. I don't know if there's a standard thread for such sensors or it's all over the place?

As poolwater eats up almost anything, how long can one reasonably expect the sensors to last?

Last, I understand some sensors come with a little circuit board with a BNC connector, I'm guessing that's a signal pre-amp to bring the sensor voltage into a measurable range? Is there any standard for sensor output ranges? Is it voltage at all or 4-20mA currentloops these things work on?

Any suggestions would be appreciated.

**Comment by rstrouse** [comment]
On my 4th year with the Atlas industrial probes.  Only calibrate once a year.

**Comment by MaxVonEvil** [comment]
Would I be correct it's one of these you have? 

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/38baa64a-96f6-4254-9494-067c512306df)

Do you use their EZO board as well?

**Reply by rstrouse** [reply]
I have the one on the left.  And an industrial ORP probe.
